drivers/soundwire/intel.c

```diff
@@ -693,6 +714,8 @@ static int intel_post_bank_switch(struct sdw_bus *bus)
 
 	ret = intel_clear_bit(shim, SDW_SHIM_SYNC, sync_reg,
 			      SDW_SHIM_SYNC_SYNCGO);
+	mutex_unlock(sdw->link_res->shim_lock);
```

Comment:
We should call mutex_unlock after line 705
```
if (!(sync_reg & SDW_SHIM_SYNC_CMDSYNC_MASK))
		return 0;
```

Reply:
yes indeed, thanks for noticing. fixed now